core/chains/evm/headtracker/heap.go

```diff
@@ -0,0 +1,33 @@
+package headtracker
+
+import evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+
+type headsHeap struct {
+	values []*evmtypes.Head
+}
+
+func (h *headsHeap) Len() int {
+	return len(h.values)
+}
+
+func (h *headsHeap) Swap(i, j int) {
+	h.values[i], h.values[j] = h.values[j], h.values[i]
+}
+
+func (h *headsHeap) Less(i, j int) bool {
+	return h.values[i].Number < h.values[j].Number
+}
+
+func (h *headsHeap) Pop() any {
+	old := h.values[len(h.values)-1]
+	h.values = h.values[:len(h.values)-1]
```

Comment:
Given the heap is structured in ascending order, shouldn't `old` pop the smallest/first item of the array?

Reply:
`headsHeap` implements `heap.Interface`, which requires Pop to remove the last element from the slice.